torch/utils/data/dataloader.py

```diff
@@ -491,7 +496,13 @@ def __init__(self, loader: DataLoader) -> None:
         self._index_sampler = loader._index_sampler
         self._num_workers = loader.num_workers
         self._prefetch_factor = loader.prefetch_factor
-        self._pin_memory = loader.pin_memory and torch.cuda.is_available()
+        # for CUDA, behaviour is default. for other backends
```

Comment:
Add expection when pin_memory_device is specified, but pin_memory is false

Reply:
added warn msg, in case pin_memory_device is set and pin_memory flag its set to false. as this a valid case.